crux_core/src/testing.rs

```diff
@@ -76,6 +76,21 @@ where
         Ok(self.context.updates())
     }
 
+    /// Resolve an effect `request` from previous update, then run the resulting event
+    ///
+    /// This helper is useful for the comment case where  one expects the effect to resolve
+    ///  to exactly one event, which should then be run by the app.
+    pub fn resolve_to_event_then_update<Op: Operation>(
```

Comment:
perhaps this should be `expect_one_event_then_update`? There's a fluent API hiding in there as well.

Reply:
I chose the word `resolve` to mirror the `Update::resolve` method. I think there is a distinction between running an event and resolving an effect.

The fluent API of something like `resolve_to_event().then_update(model)` might be there somewhere but as written the types won't line up...